### PR TITLE
feat: diff github annotation format

### PR DESCRIPTION
**Pull request by goatwu1993**
Implement
`pycobertura diff --format github-annotation`

Motivation:
As a code reviewer, sometimes it is useful to see uncovered lines in PR with only changes relative to base_ref

### CODE REVIEW

File: tests/test_reporters.py [hunk 1]
```diff
@@ -705,3 +705,43 @@ def test_github_annotation_report(
         )
         == expected_custom_output
     )
+
+
+@pytest.mark.parametrize(
+    "report1, report2, expected_default_output, expected_custom_output",
```
Comment (aconrad) [inline tests/test_reporters.py:711]: Can we write 2 parametrized tests, one for the default config and one for the custom config? Something like:

```python
"report1, report2, config, expected_default_output, expected_custom_output"
```

... and then pass the config dict as parametrized input to the test.

**Comment by aconrad** [comment]
@goatwu1993 thanks so much for this PR, it's great!

File: tests/test_reporters.py (hunk 1 above)
Comment (goatwu1993) [reply tests/test_reporters.py]: [@fce](https://github.com/aconrad/pycobertura/pull/166/commits/feb16f4e45ffb33f0ff179e575846b49adca4e26)

**Review by aconrad — APPROVED** [review]
Thanks!

**Comment by aconrad** [comment]
[v3.2.0](https://pypi.org/project/pycobertura/3.2.0/)

I'm not sure why the github-annotation screenshots are broken on pypi as well as others, but not all... 🤔

**Comment by goatwu1993** [comment]
@aconrad Hmm, no idea as well. I tried release.sh (without twine upload). unpack the tar.gz the image is still openable. No idea what the black magic behind pypi...

The PRs which I am testing/screenshotting github-annotation show/diff remained opened...
https://github.com/goatwu1993/pycobertura-test/pull/6
https://github.com/goatwu1993/pycobertura-test/pull/7

**Pull request merged**